proposals/0010-resource-element-type-validation.md

```diff
@@ -0,0 +1,217 @@
+* Proposal: [0010](0010-resource-element-type-validation.md)

```

Comment:
nit: I think we already have a proposal 10, so make sure to update the number before merging.